fem/tmop_tools.cpp

```diff
@@ -218,6 +218,73 @@ void ParAdvectorCGOper::Mult(const Vector &ind, Vector &di_dt) const
 }
 #endif
 
+#ifdef MFEM_USE_GSLIB
+void InterpolatorFP::SetInitialField(const Vector &init_nodes,
+                                     const Vector &init_field)
+{
+   nodes0 = init_nodes;
+   field0 = init_field;
+   Mesh *m = mesh;
+#ifdef MFEM_USE_MPI
+   if (pmesh) { m = pmesh; }
+#endif
+   m->SetNodes(nodes0);
+
+   const double rel_bbox_el = 0.1;
+   const double newton_tol  = 1.0e-12;
+   const int npts_at_once   = 256;
+
+   FiniteElementSpace *f = fes;
+#ifdef MFEM_USE_MPI
+   if (pmesh) {finder = new FindPointsGSLIB(pfes->GetComm());}
+   else       {finder = new FindPointsGSLIB();}
+   if (pfes)  {f = pfes;}
+#else
+   finder = new FindPointsGSLIB();
+#endif
+   finder->Setup(*m, rel_bbox_el, newton_tol, npts_at_once);
+
+   field0_gf.SetSpace(f);
+   for (int i = 0; i < field0.Size(); i++)
+   {
+      field0_gf(i) = field0(i);
```

Comment:
You can copy into `field0_gf` directly from `init_field`. and eliminate the variable `field0.`